master/internal/proxy/proxy_intg_test.go

```diff
@@ -98,3 +102,53 @@ func TestProxyLifecycle(t *testing.T) {
 	}
 	require.Equal(t, 0, len(DefaultProxy.Summary()))
 }
+
+func TestNewProxyHandler(t *testing.T) {
+	// First init the new Proxy
+	InitProxy(proxyAuth)
+	// And check that the Proxy struct is set up correctly
+	require.NotNil(t, DefaultProxy.HTTPAuth)
+	require.Equal(t, map[string]*Service{}, DefaultProxy.services)
+	require.Equal(t, "", DefaultProxy.syslog.Message)
+
+	e := echo.New()
+	// Create http test recorder
+	req := httptest.NewRequest(http.MethodGet, u.Path+"/proxy", nil)
+	// Create new echo context
+	c := e.NewContext(req, httptest.NewRecorder())
+
+	go func() {
+		if err := e.Start(u.Path); err != nil && err != http.ErrServerClosed {
+			t.Logf("failed to start server: %s", err)
+		}
+	}()
+
+	// Ensure server is up, before testing it.
+	i := 0
+	tk := time.NewTicker(time.Second)
+	defer tk.Stop()
+	for range tk.C {
+		resp, err := http.Get("http://" + u.Path + "/proxy")
+		if err == nil {
+			resp.Body.Close() //nolint:errcheck
+			break
+		}
+		i++
+		if i > 5 {
+			t.FailNow()
+		}
+	}
```

Comment:
i have a function, `waitForCondition`, i have been copying around. want to just use that (copied) and i'll add it to test utils soon?
func: https://github.com/determined-ai/determined/pull/7195/files#diff-8c94b431f1c34c1a7d2610f2a51ca568294806e28731083bc4da5c8b495d0941R198-R208
usage: https://github.com/determined-ai/determined/pull/7195/files#diff-8c94b431f1c34c1a7d2610f2a51ca568294806e28731083bc4da5c8b495d0941R187

Reply:
added!